From f0b51f4ffd469bcf5eab666c4c9373a7479e888a Mon Sep 17 00:00:00 2001
From: Daniel Cederman <cederman@gaisler.com>
Date: Mon, 15 Jun 2020 09:05:34 +0200
Subject: [PATCH 14/50] common: Upload all gcda files from tests with multiple
 source files

By default only the gcda file from the main test source file is
uploaded.
---
 gcc/testsuite/lib/profopt.exp | 24 ++++++++++++++++++------
 1 file changed, 18 insertions(+), 6 deletions(-)

diff --git a/gcc/testsuite/lib/profopt.exp b/gcc/testsuite/lib/profopt.exp
index 0b853a1559f..466b7c8b2c9 100644
--- a/gcc/testsuite/lib/profopt.exp
+++ b/gcc/testsuite/lib/profopt.exp
@@ -312,6 +312,7 @@ proc profopt-execute { src } {
     global generate_final_code use_final_code
     global verbose
     global testname_with_flags
+    global additional_sources_used
 
     if ![info exists profile_option] {
         error "No profile option specified for first compile."
@@ -460,12 +461,23 @@ proc profopt-execute { src } {
 	    }
 
 	    foreach ext $prof_ext {
-		remote_upload target $tmpdir/$bprefix$base.$ext
-		set files [glob -nocomplain $bprefix$base.$ext]
-		if { $files == "" } {
-		    set status "fail"
-		    set missing_file 1
-		    fail "$testcase execution: file $bprefix$base.$ext does not exist, $option $profile_option"
+		set prof_files $bprefix$base.$ext
+		if [info exists additional_sources_used] {
+		    foreach srcfile $additional_sources_used {
+			set add_basename [file tail $srcfile]
+			set add_base [file rootname $add_basename]
+			lappend prof_files $bprefix$add_base.$ext
+		    }
+		}
+
+		foreach prof_file $prof_files {
+		    remote_upload target $tmpdir/$prof_file
+		    set files [glob -nocomplain $prof_file]
+		    if { $files == "" } {
+			set status "fail"
+			set missing_file 1
+			fail "$testcase execution: file $prof_file does not exist, $option $profile_option"
+		    }
 		}
 	    }
 	}
-- 
2.34.1

